src/libcore/intrinsics.rs

```diff
@@ -1331,6 +1333,26 @@ extern "rust-intrinsic" {
 // (`transmute` also falls into this category, but it cannot be wrapped due to the
 // check that `T` and `U` have the same size.)
 
+/// Checks whether `ptr` is properly aligned with respect to
+/// `align_of::<T>()`.
+pub(crate) fn is_aligned_and_not_null<T>(ptr: *const T) -> bool {
+    return !ptr.is_null() && ptr as usize % mem::align_of::<T>() == 0;
+}
+
+/// Checks whether the regions of memory starting at `src` and `dst` of size
+/// `count * size_of::<T>()` overlap.
+fn overlaps<T>(src: *const T, dst: *const T, count: usize) -> bool {
+    let src_usize = src as usize;
+    let dst_usize = dst as usize;
+    let size = mem::size_of::<T>().checked_mul(count).unwrap();
+    let diff = if src_usize > dst_usize {
```

Comment:
I'm surprised there's no utility function in libcore for this... should there be one?

Reply:
For what? The absolute difference of two unsigned integers?

I don't know if there should be one. ;) Where would it even be?

Reply:
Yeah the absolute diff... Somewhere in `core::num` or as methods on the types?

Reply:
Hm, not sure if this PR is the right place to introduce some new unstable public API?

Reply:
Yeah I agree, it was an aside remark, sorry should have been more clear :)

Reply:
Filed as https://github.com/rust-lang/rust/issues/62111.

Reply:
Thanks!